controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1854,6 +1854,52 @@ func (r *BareMetalHostReconciler) createHostFirmwareSettings(info *reconcileInfo
 
 	return nil
 }
+// NOTE(janders) start HostUpdatePolicy mods
+
+func (r *BareMetalHostReconciler) createHostUpdatePolicy(info *reconcileInfo) error {
+	// Check if HostFirmwareComponents already exists
+	hup := &metal3api.HostUpdatePolicy{}
+	if err := r.Get(info.ctx, info.request.NamespacedName, hup); err != nil {
+		if k8serrors.IsNotFound(err) {
+			// A resource doesn't exist, create one
+			hup.ObjectMeta = metav1.ObjectMeta{
+				Name:      info.host.Name,
+				Namespace: info.host.Namespace}
+
+			hup.Spec = metal3api.HostUpdatePolicySpec{
+				FirmwareSettings: metal3api.UpdatePolicy{}, 
+				FirmwareUpdates: metal3api.UpdatePolicy{}}
```

Comment:
This is clearly wrong - running tests results in:
```
janders@darkstar:~/git/hostupdatepolicy/baremetal-operator$ go test
# github.com/metal3-io/baremetal-operator/controllers/metal3.io
controllers/metal3.io/baremetalhost_controller.go:1870:23: invalid composite literal type "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1".UpdatePolicy
controllers/metal3.io/baremetalhost_controller.go:1871:22: invalid composite literal type "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1".UpdatePolicy
FAIL	github.com/metal3-io/baremetal-operator [build failed]
janders@darkstar:~/git/hostupdatepolicy/baremetal-operator$ 
```
likely missing something obvious - any pointers appreciated! :)